### DISCUSSION TITLE
How to make left drawer responsive

### DISCUSSION THREAD

**Discussion by jixian01** [zauberzeug/nicegui#2061] [Q&A] [answered]
Hello all,
i am new to niceGUI and have a question: how to make left drawer responsive to different screen size?
i tried with classes but no success. 
`ui.left_drawer(top_corner=False, bottom_corner=True, fixed=False).classes('md:w-100 lg:w-200') `

thank you in advance for any help!

**Comment by falkoschindler** [comment, marked as answer]
Hi @jixian01,
The classes "md:w-100" and "lg:w-200" are invalid, because Tailwind only offers certain width values per default (see https://tailwindcss.com/docs/width). Using custom values with square brackets like "lg:w-[200px]" _should_ work though - but it doesn't. Somehow it doesn't seem to affect the drawer element.

Luckily we can address the `<aside>` tag directly with some custom CSS:

```py
ui.add_head_html('''
    <style>
        aside.q-drawer {
            width: 200px !important;
        }
        @media (max-width: 1200px) {
            aside.q-drawer {
                width: 100px !important;
            }
        }
    </style>
''')
ui.left_drawer(top_corner=False, bottom_corner=True, fixed=False, bordered=True)
```

Of course, you can adjust the widths and the breakpoint as you wish.

**Reply by jixian01** [reply]
thank you @falkoschindler!
i tried this and it works. Thank you!
Is there any document about how styling works in nicegui. I went through the official document, there are several possibilities to do styling: props, classes, styles and some from quasar, some could from tailwind and css. But i can't find any information about the priority in case of conflict, and also no information about which tailwind classes work/not work for which element (like the case above).

**Reply by falkoschindler** [reply]
No, this isn't documented in great detail. It's probably not feasible to foresee and document all possible Tailwind classes and CSS properties for all possible elements in different situations. But we should definitely add some guidelines on how to debug and solve such layout and styling problems.

**Comment by natankeddem** [comment]
Quasar drawer includes a responsive behavior with a mini mode. I haven't tried utilizing, it has anyone else tried?
